pkg/core/resources/apis/mesh/mesh_validator.go

```diff
@@ -165,8 +167,12 @@ func validateDatadog(cfgStr *structpb.Struct) validators.ValidationError {
 		verr.AddViolation("address", "cannot be empty")
 	}
 
-	if cfg.Port > 0xFFFF || cfg.Port < 1 {
-		verr.AddViolation("port", "must be in the range 1 to 65535")
+	if !strings.HasPrefix(cfg.Address, "unix:") {
```

Comment:
Suggest using `url.Parse` and looking at the scheme.

Reply:
@jpeach I initially started writing this with `url.Parse`. It really doesn't play nicely with the unix domain socket scheme though, as it interprets the path as either "/host/path" or simply "/path", depending on the initial number of slashes.  I always try to use existing libraries and utils rather than writing my own, but in this case it turned out it I'd still be using the same amount of conditional string parsing on the results.

Also scheme is required, so that "127.0.0.1:8126" will not work. It would have to be "http://127.0.0.1:8126", which seemed odd to me since the input is really host and port, not a full URL, and "http", "https", or any other non-`unix://` scheme is irrelevant and possibly incorrect. Finally, `url.Parse` doesn't even correctly validate IPv6, which was kind of the last chance I gave it for making this code simpler or more robust..

Reply:
> @jpeach I initially started writing this with `url.Parse`. It really doesn't play nicely with the unix domain socket scheme though, as it interprets the path as either "/host/path" or simply "/path", depending on the initial number of slashes. I always try to use existing libraries and utils rather than writing my own, but in this case it turned out it I'd still be using the same amount of conditional string parsing on the results.
> 
> Also scheme is required, so that "127.0.0.1:8126" will not work. It would have to be "http://127.0.0.1:8126", which seemed odd to me since the input is really host and port, not a full URL, and "http", "https", or any other non-`unix://` scheme is irrelevant and possibly incorrect. Finally, `url.Parse` doesn't even correctly validate IPv6, which was kind of the last chance I gave it for making this code simpler or more robust..

Yeh, it looks like the scheme is required for IPv6 literals, but not for IPv6. I agree that if we can't just traffic in URLs here, then it's probably not worth it.

One thing that occurs to me is that we don't really need to require a "unix:" prefix, we can just look for the leading '/'. So anything that begins with '/' would be a socket path.

Reply:
Hmm, yeah, in fact since we're not using URLs, anything which contains any '/' at all is not hostname:port, and therefore is a path. Although, I don't know what a relative path would mean in this instance.

Reply:
If we allow relative paths, it would allow people to depend on the current working directory. Suggest that that we require an absolute path here.